dev/src/watch.ts

```diff
@@ -252,7 +252,8 @@ abstract class Watch<T = DocumentData> {
 
     this.initStream();
 
-    return () => {
+    const unsubscribe = () => {
+      this.firestore.unregisterListener(this.requestTag);
```

Comment:
I don't think this is the only place that Listens are terminated. What happens if we throw an error and terminate from our end?

Reply:
`this.isActive` is only set here and in `closeStream()` (not counting constructor initialization). Added an unregister call in both places iff `isActive` is set to true.